logstash-core/lib/logstash/instrument/null_metric.rb

```diff
@@ -2,41 +2,39 @@
 require "logstash/instrument/metric"
 
 module LogStash module Instrument
+ # This class is used in the context when we disable the metric collection
+ # for specific plugin to replace the `NamespacedMetric` class with this one
+ # which doesn't produce any metric to the collector.
  class NullMetric
-   class NullTimedExecution
-     def self.stop
-     end
-   end
-
-   # Allow to reuse the same variable when creating subnamespace
-   NULL_METRIC_INSTANCE = NullMetric.new
+    attr_reader :namespace_name
 
-   attr_reader :collector, :namespace_information
-   def initialize
+   # Get only the instance methods defined in the class
+   # and not the whole hierarchy of methods.
+   LogStash::Instrument::Metric.public_instance_methods(false).each do |method|
```

Comment:
and in the other side it makes readability much harder, I wonder if this method is actually faster or slower than pure composite here. What do you think?